id-repository/id-repository-identity-service/src/main/java/io/mosip/idrepository/identity/helper/IdRepoServiceHelper.java

```diff
@@ -0,0 +1,165 @@
+     */
+    public Map<String, Object> convertToMap(Object identity) throws IdRepoAppException {
+        try {
+            return mapper.readValue(mapper.writeValueAsBytes(identity), new TypeReference<Map<String, Object>>() {
+            });
+        } catch (IOException e) {
+            mosipLogger.error(IdRepoSecurityManager.getUser(), ID_REPO_SERVICE_HELPER, "convertToMap", "\n" + e.getMessage());
+            throw new IdRepoAppException(INVALID_INPUT_PARAMETER.getErrorCode(),
+                    String.format(INVALID_INPUT_PARAMETER.getErrorMessage(), REQUEST), e);
+        }
+    }
+
+    /**
+     * Gets the schema.
+     *
+     * @param schemaVersion the schema version
+     * @return the schema
+     */
+    public String getSchema(String schemaVersion) {
+        if (Objects.isNull(schemaVersion) || schemaVersion.contentEquals("null")) {
+            mosipLogger.error(IdRepoSecurityManager.getUser(), ID_REPO_SERVICE_HELPER, "getSchema",
+                    "\n" + "schemaVersion is null");
+            throw new IdRepoAppUncheckedException(MISSING_INPUT_PARAMETER.getErrorCode(),
+                    String.format(MISSING_INPUT_PARAMETER.getErrorMessage(),
+                            ROOT_PATH + IdObjectValidatorConstant.PATH_SEPERATOR + "IDSchemaVersion"));
+        }
+        try {
+            if (schemaMap.containsKey(schemaVersion)) {
+                return schemaMap.get(schemaVersion);
+            } else {
+                RestRequestDTO restRequest;
+                restRequest = restBuilder.buildRequest(RestServicesConstants.SYNCDATA_SERVICE, null, ResponseWrapper.class);
+                restRequest.setUri(restRequest.getUri().concat("?schemaVersion=" + schemaVersion));
+                ResponseWrapper<Map<String, String>> response = restHelper.requestSync(restRequest);
+                schemaMap.put(schemaVersion, response.getResponse().get("schemaJson"));
+                supportedHandlesInSchema.put(schemaVersion, getSupportedHandles(response.getResponse().get("schemaJson")));
+                return getSchema(schemaVersion);
+            }
+        } catch (IdRepoDataValidationException | RestServiceException e) {
+            mosipLogger.error(IdRepoSecurityManager.getUser(), ID_REPO_SERVICE_HELPER, "getSchema", "\n" + e.getMessage());
+            throw new IdRepoAppUncheckedException(IdRepoErrorConstants.SCHEMA_RETRIEVE_ERROR);
+        }
+    }
+
+    public Map<String, HandleDto> getSelectedHandles(Object identity) throws IdRepoAppException {
+        Map<String, Object> requestMap = convertToMap(identity);
+        if (requestMap.containsKey(ROOT_PATH) && Objects.nonNull(requestMap.get(ROOT_PATH))) {
+            Map<String, Object> identityMap = (Map<String, Object>) requestMap.get(ROOT_PATH);
+            String schemaVersion = String.valueOf(identityMap.get(ID_SCHEMA_VERSION));
+            if(identityMap.containsKey(SELECTED_HANDLES) && Objects.nonNull(identityMap.get(SELECTED_HANDLES))) {
```

Comment:
keys for id schema and selected handles need to be taken from identity mapping and then queried into the identity object

Reply:
Done